src/query/storages/system/src/query_log_table.rs

```diff
@@ -367,6 +376,27 @@ impl SystemLogElement for QueryLogElement {
             .next()
             .unwrap()
             .push(Scalar::String(self.query_text.clone()).as_ref());
+        let query_hash = format!("{:x}", Md5::digest(&self.query_text));
+        columns
+            .next()
+            .unwrap()
+            .push(Scalar::String(query_hash).as_ref());
+        let tokens = tokenize_sql(&self.query_text)?;
+        let query_parameterized_hash = parse_sql(&tokens, Dialect::PostgreSQL)
+            .ok()
+            .and_then(|(stmt, _)| {
+                if let Statement::Query(_) = stmt {
```

Comment:
in stream load, the insert sql_text maybe like this : query_text: INSERT INTO products (id, name, description) VALUES (?,?,?)

If try to format this statement, will return None in format_statement pop:

https://github.com/datafuselabs/databend/blob/05e7d49be157a6aad8ef4549d4164e840e5e204b/src/query/ast/src/ast/format/ast_format.rs#L28

So I think we can only record the select query in here.

But also I think we can modify the format_statement like this:

```rust
pub fn format_statement(stmt: Statement, ignore_literal: bool) -> Result<String> {
    let mut visitor = AstFormatVisitor::new(ignore_literal);
    visitor.visit_statement(&stmt);
    match visitor.children.pop() {
        Some(format_ctx) => format_ctx.format_pretty(),
        None => Err(ErrorCode::SyntaxException(format!("Format stmt {} None", stmt)))
    }
}

```

cc @b41sh @andylokandy

Reply:
we can ignore values clause here.

Reply:
Refactor in new commit